src/Analyzers/CSharp/CodeFixes/MakeLocalFunctionStatic/MakeLocalFunctionStaticCodeFixHelper.cs

```diff
@@ -82,7 +86,9 @@ internal static class MakeLocalFunctionStaticCodeFixHelper
                 }
             }
 
-            var parameterAndCapturedSymbols = CreateParameterSymbols(captures);
+            var thisParameter = (IParameterSymbol?)captures.FirstOrDefault(c => c.IsThisParameter());
+
+            var parameterAndCapturedSymbols = CreateParameterSymbols(captures.Where(c => !c.IsThisParameter()));
```

Comment:
💡 If you use `WhereAsArray`, there are no allocations at all in the case where the selection produces all or none.

Reply:
quite right.  Done :)